### PR TITLE
Add the mailto: example

### PR DESCRIPTION
**Pull request by ivan-diachenko-td**
Now there is no possibility to open a desktop mail app by clicking `mailto:` link in JxBrowser. This PR suggests the workaround on how to open a default mail client in the environment using `Desktop` class.

### CODE REVIEW

File: network/src/main/java/MailTo.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import java.awt.event.WindowEvent;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static com.teamdev.jxbrowser.engine.RenderingMode.HARDWARE_ACCELERATED;
+import static java.awt.Desktop.isDesktopSupported;
+
+/**
+ * This example demonstrates how to open a default desktop mail client by clicking {@code mailto:} links.
```
Comment (vladimir-ikryanov) [inline network/src/main/java/MailTo.java:41]: The idea of this example is not to show how to open a default desktop mail client. It should explain how to handle the case when user clicks a link with the "mailto:" URL and expects that a default desktop mail client is opened with the target email address, so they can send an email.

**Comment by ivan-diachenko-td** [comment]
@vladimir-ikryanov @vlad-lubenskyi PTAL

**Review by vladimir-ikryanov — CHANGES_REQUESTED** [review]

File: network/src/main/java/MailTo.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            Desktop desktop = Desktop.getDesktop();
+            if (isDesktopSupported() && desktop.isSupported(Desktop.Action.MAIL)) {
+                try {
+                    UrlRequest urlRequest = params.urlRequest();
+                    String url = urlRequest.url();
+                    if (url.contains("mailto:")) {
+                        desktop.mail(new URI(url));
+                    }
+                } catch (URISyntaxException | IOException e) {
+                    throw new RuntimeException(e);
```
[the remark this answers is not in the document]
Comment (ivan-diachenko-td) [reply network/src/main/java/MailTo.java]: Changed to `printStackTrace`

**Comment by ivan-diachenko-td** [comment]
@vladimir-ikryanov PTAL

**Review by vladimir-ikryanov — APPROVED** [review]
LGTM

File: network/src/main/java/HandleMailToClick.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    public static void main(String[] args) {
+        Engine engine = Engine.newInstance(
+                EngineOptions.newBuilder(HARDWARE_ACCELERATED).build());
+        Browser browser = engine.newBrowser();
+
+        engine.network().set(BeforeUrlRequestCallback.class, params -> {
+            String url = params.urlRequest().url();
+            if (url.startsWith("mailto:")) {
+                Desktop desktop = Desktop.getDesktop();
+                if (isDesktopSupported() && desktop.isSupported(Desktop.Action.MAIL)) {
```
Comment (anna-dolbina) [inline network/src/main/java/HandleMailToClick.java:56]: Consider checking `isDesktopSupported` before calling `Desktop.getDesktop()`. In other case, you may get an `UnsupportedOperationException`:
https://docs.oracle.com/javase/9/docs/api/java/awt/Desktop.html#getDesktop--

**Pull request merged**